src/metal.jl

```diff
@@ -774,3 +746,283 @@ function add_module_metadata!(@nospecialize(job::CompilerJob), mod::LLVM.Module)
+
+                    # then, check if both arguments are zero and have a mismatching sign.
+                    # if so, return in accordance to the intrinsic (minimum or maximum)
+
+                    position!(builder, bb_check_zero)
+
+                    typ′ = LLVM.IntType(8*sizeof(jltyp); ctx)
+                    arg0′ = bitcast!(builder, arg0, typ′)
+                    arg1′ = bitcast!(builder, arg1, typ′)
+
+                    arg0_zero = fcmp!(builder, LLVM.API.LLVMRealUEQ, arg0,
+                                      LLVM.ConstantFP(typ, zero(jltyp)))
+                    arg1_zero = fcmp!(builder, LLVM.API.LLVMRealUEQ, arg1,
+                                      LLVM.ConstantFP(typ, zero(jltyp)))
+                    args_zero = and!(builder, arg0_zero, arg1_zero)
+                    arg0_sign = and!(builder, arg0′, LLVM.ConstantInt(typ′, Base.sign_mask(jltyp)))
+                    arg1_sign = and!(builder, arg1′, LLVM.ConstantInt(typ′, Base.sign_mask(jltyp)))
+                    sign_mismatch = icmp!(builder, LLVM.API.LLVMIntNE, arg0_sign, arg1_sign)
+                    relevant_zero = and!(builder, args_zero, sign_mismatch)
+                    br!(builder, relevant_zero, bb_compare_zero, bb_fallback)
+
+                    position!(builder, bb_compare_zero)
+                    arg0_negative = icmp!(builder, LLVM.API.LLVMIntNE, arg0_sign,
+                                          LLVM.ConstantInt(typ′, 0))
+                    val = if intr == LLVM.Intrinsic("llvm.minimum")
+                        select!(builder, arg0_negative, arg0, arg1)
+                    else
+                        select!(builder, arg0_negative, arg1, arg0)
+                    end
+                    ret!(builder, val)
+
+                    # finally, it's safe to use the existing minnum/maxnum intrinsics
+
+                    position!(builder, bb_fallback)
+                    fallback_intr_fn = if intr == LLVM.Intrinsic("llvm.minimum")
+                        "air.fmin.f$(8*sizeof(jltyp))"
+                    else
+                        "air.fmax.f$(8*sizeof(jltyp))"
+                    end
+                    fallback_intr = if haskey(functions(mod), fallback_intr_fn)
+                        functions(mod)[fallback_intr_fn]
+                    else
+                        LLVM.Function(mod, fallback_intr_fn, LLVM.FunctionType(typ, parameters(call_ft)))
+                    end
+                    val = call!(builder, fallback_intr, collect(parameters(new_intr)))
+                    ret!(builder, val)
+                end
+            end
+
+            @dispose builder=IRBuilder(ctx) begin
+                position!(builder, call)
+                debuglocation!(builder, call)
+
+                new_value = call!(builder, new_intr, arguments(call))
+                replace_uses!(call, new_value)
+                unsafe_delete!(bb, call)
+                changed = true
+            end
+        end
```

Comment:
@gbaraldi Look what you made me do.

Reply:
Sorry :(. It's very annoying that backends don't know how to lower this.

Reply:
No problem 🙂 But yeah it is annoying that you can't reasonably expect fairly basic things to be available across all back-ends; it should be easy enough for LLVM to provide a fallback lowering...

Reply:
X86 not having a lowering is telling though.